### ISSUE TITLE
Unexpected behavior when editing an integration while in building process

### ISSUE THREAD

**Issue by mcada** [syndesisio/syndesis#3095]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

If you edit integration when its in building state and delete any step, it will reappear again.

## Tasks involved / Steps to Reproduce

1. Create an integration (I tested it with database --> log --> database)
2. Click on publish
3. Edit the integration while its building
4. Delete one step
5. The step will reappear after few seconds

![out](https://user-images.githubusercontent.com/14313995/42817802-8149e124-89cf-11e8-8e5d-a937b7736145.gif)

Sometimes it breaks in different way and this happens:
![out](https://user-images.githubusercontent.com/14313995/42817739-491518f0-89cf-11e8-862a-53adbdc2988a.gif)

**Comment by gashcrumb** [comment]
Hmm, this is due to the integration being fetched as the UI is notified of changes, causing the step array to be reloaded.  I can probably do some comparison and prevent this from happening though.

The *only* drawback I see is if 2 browsers have an integration open, one makes changes and saves it, the other browser wouldn't see that change unless they exit the editor and go back in.

**Comment by dongniwang** [comment]
> So if we want that behavior should we disable the edit button while the integration is publishing. 

This was my first thought as well. Any foreseeable drawbacks?

**Comment by gashcrumb** [comment]
*shrugs*

**Comment by sjcox-rh** [comment]
@gashcrumb @dongniwang agreed. Since editing while an integration is in the publish state can potentially cause problems, why allow them to edit? +1 for disabling. 

Unless I'm missing use cases where editing capability is necessary while publishing. Only concern is if user tries to go into edit mode and is confused as to why the button is disabled. Perhaps a tooltip explaining?

**Comment by gashcrumb** [comment]
Will disable the edit links while an integration is pending as part of the work for #3180
